packages/x-components/src/x-modules/facets-next/components/lists/__tests__/filters-injection.mixin.spec.ts

```diff
@@ -0,0 +1,189 @@
+      async setPropFilters(propFilters: Filter[]) {
+        await wrapper.setData({ propFilters });
+      }
+    };
+  }
+
+  it('renders the props filters', async () => {
+    const whiteFilter = createNextSimpleFilter('color', 'white');
+    const blackFilter = createNextSimpleFilter('color', 'black');
+    const blueFilter = createNextSimpleFilter('color', 'blue');
+    const { setPropFilters, getRenderedFilterIds } = renderFiltersMixin({
+      propFilters: [whiteFilter, blackFilter]
+    });
+    let renderedIds = getRenderedFilterIds();
+
+    expect(renderedIds).toContain(whiteFilter.id);
+    expect(renderedIds).toContain(blackFilter.id);
+
+    await setPropFilters([whiteFilter, blackFilter, blueFilter]);
+    renderedIds = getRenderedFilterIds();
+
+    expect(renderedIds).toContain(whiteFilter.id);
+    expect(renderedIds).toContain(blackFilter.id);
+    expect(renderedIds).toContain(blueFilter.id);
+  });
+
+  it('renders the injected filters', () => {
+    const whiteFilter = createNextSimpleFilter('color', 'white');
+    const blackFilter = createNextSimpleFilter('color', 'black');
+    const blueFilter = createNextSimpleFilter('color', 'blue');
+    const { getRenderedFilterIds } = renderFiltersMixin({
+      injectedFilters: [whiteFilter, blackFilter, blueFilter]
+    });
+    let renderedIds = getRenderedFilterIds();
+
+    expect(renderedIds).toContain(whiteFilter.id);
+    expect(renderedIds).toContain(blackFilter.id);
+    expect(renderedIds).toContain(blueFilter.id);
+  });
+
+  it('renders the prop filters when there are injected filters too', () => {
+    const whiteFilter = createNextSimpleFilter('color', 'white');
+    const blackFilter = createNextSimpleFilter('color', 'black');
+    const blueFilter = createNextSimpleFilter('color', 'blue');
+    const { getRenderedFilterIds } = renderFiltersMixin({
+      propFilters: [whiteFilter, blackFilter],
+      injectedFilters: [blueFilter]
+    });
+    let renderedIds = getRenderedFilterIds();
+
+    expect(renderedIds).toContain(whiteFilter.id);
+    expect(renderedIds).toContain(blackFilter.id);
+    expect(renderedIds).not.toContain(blueFilter.id);
+  });
+
+  it('renders the first level Hierarchical Filters when no parentId prop passed', () => {
+    /*
+     Women
+     Men
+       Jeans
+        Shirts
+     */
+    const categoryShirts = createNextHierarchicalFilter('category', 'shirts', false);
+    const categoryJeans = createNextHierarchicalFilter('category', 'jeans', false);
+    const categoryMen = createNextHierarchicalFilter('category', 'men', false, [
+      categoryShirts.id,
+      categoryJeans.id
+    ]);
+    categoryShirts.parentId = categoryJeans.parentId = categoryMen.id;
+    const categoryWomen = createNextHierarchicalFilter('category', 'women', false);
+
+    const { getRenderedFilterIds } = renderFiltersMixin({
+      propFilters: [categoryShirts, categoryJeans, categoryMen, categoryWomen]
+    });
+    let renderedIds = getRenderedFilterIds();
```

Comment:
```suggestion
    const renderedIds = getRenderedFilterIds();
```

- [ ] Change to `const`